src/Orleans/Async/TaskExtensions.cs

```diff
@@ -216,7 +216,7 @@ internal static Task<T> ConvertTaskViaTcs<T>(Task<T> task)
                     {
                         resolver.TrySetResult(t.Result);
                     }
-                });
+                }, TaskContinuationOptions.ExecuteSynchronously);
```

Comment:
I wonder if all parts of `ConvertTaskViaTcs` are necessary. It is solely used in `InvokeMethodAsync` path.

We can inline the method and remove everything other than:
```c#
if (task == null) return Task.FromResult(default(T));
```
and
```c#
if (task.Status == TaskStatus.Created) task.Start();
```